app/assets/javascripts/mobile/mobile_drawer.js

```diff
@@ -1,20 +1,147 @@
+      var hammerOpts = {
+        "prevent_default": false,
+        threshold: 0,
+        direction: Hammer.DIRECTION_HORIZONTAL
+      };
+
       $("#all_aspects").bind("tap click", function(evt){
         evt.preventDefault();
         $(this).find("+ li").toggleClass("hide");
       });
 
-      $("#menu-badge").bind("tap click", function(evt){
+      $("#followed_tags").bind("tap click", function(evt){
         evt.preventDefault();
-        $("#app").toggleClass("draw");
+        $(this).find("+ li").toggleClass("hide");
       });
 
-      $("#followed_tags").bind("tap click", function(evt){
+      $("#menu-badge").on("tap click", function(evt){
         evt.preventDefault();
-        $(this).find("+ li").toggleClass("hide");
+        self.createOverlay();
+        self.openDrawer();
       });
+
+      this.mainContent.on("tap click", function(){
+        if(self.isDrawerOpened()){
+          self.closeDrawer();
+        }
+      });
+
+      this.dragZone.hammer(hammerOpts).bind("panstart", function(){
+          self.createOverlay();
+          $("body").css("overflow", "hidden");
+          self.drawer.addClass("gesture").removeClass("closed-drawer").removeClass("opened-drawer");
+        })
+        .bind("pan", function(e){
+          if(Date.now() - dateLastEvent < debouncingThreshold){
+            return e.preventDefault();
+          }
+
+          if(e.gesture.pointerType === "touch"){
+            dateLastEvent = Date.now();
+            var distanceFromRight = window.innerWidth - e.gesture.center.x;
+
+            // Keep within boudaries
+            distanceFromRight = Math.max(distanceFromRight, 0);
+            distanceFromRight = Math.min(distanceFromRight, self.drawer.width());
+
+            self.drawer.css("left", "calc(100% - " + distanceFromRight + "px)");
+            $("#main-overlay").css("opacity", distanceFromRight / self.drawer.width());
+          }
+        })
+        .bind("panend", function(e){
+          if(e.gesture.pointerType === "touch"){
+            var velocityX = e.gesture.velocityX;
+            var openedMenu = (window.innerWidth - e.gesture.center.x) < (window.innerWidth / 2);
+            // Open the menu
+            if(!openedMenu || velocityX > 0.3){
+              self.openDrawer();
+            }
+            // Close the menu
+            else if((openedMenu && velocityX <= 0.3) || velocityX < -0.3){
+              self.closeDrawer();
+            }
+          }
+        });
+    },
+
+    createDragZone: function(){
+      var dragZone = $("<div id='drag-zone'></div>");
+      dragZone.css({
+        position: "fixed",
+        top: 0,
+        right: 0,
+        height: "100vh",
+        width: "15px",
+        "z-index": 100
+      });
+      return dragZone;
+    },
+
+    mutateDragZone: function(){
+      if(this.isDrawerOpened()){
+        this.dragZone.css({
+          width: window.innerWidth - (this.drawer.width() - 40),
+          right: "unset",
```

Comment:
- Expected space(s) after "else".
- Closing curly brace does not appear on the same line as the subsequent block.